app/workers/order_worker.rb

```diff
@@ -0,0 +1,8 @@
+class OrderWorker
```

Comment:
Tìm hiểu ActiveJob và Redis worker

Reply:
Active Jobs là một framework cho việc thiết lập các jobs và làm chúng có thể chạy trên nhiều backends. Hiểu đơn giản hơn là nó sẽ tạo ra 1 luồng riêng giống trong Java để thực hiện một công việc khác mà không ảnh hưởng đến luồng chính. Điều đó làm tăng trải nghiệm người dùng.
về cơ bản thì Worker và job giống nhau. ActiveJob gọi nó là một Job trong khi Sidekiq gọi nó là một Worker. Tuy nhiên,ActiveJob không cung cấp quyền truy cập vào tập hợp đầy đủ các tùy chọn Sidekiq, vì vậy nếu muốn tùy chỉnh các tùy chọn cho công việc của mình, thì ta có thể cần phải biến nó thành Worker.
ActiveJob hoạt động trên object model, có nghĩa có thể truyền 1 object model là tham số của method perform
còn Worker thì ko, chỉ nhận các tham số pure ruby
Tuy nhiên, sidekiq hơn về Performance: Active Job is 2-20x times slower pushing jobs to Redis and has ~3x the processing overhead (with Rails 5.1.4 and Sidekiq 5.1.1).